samples/junit-tests/src/test/java/com/yelp/codegen/generatecodesamples/PostFileEndpointTest.kt

```diff
@@ -0,0 +1,179 @@
+import org.junit.Assert.assertTrue
+import org.junit.Rule
+import org.junit.Test
+
+private data class FileApiStats(
+    var boundary: String? = null,
+    var errors: List<String>? = null,
+    var parts: List<MultiPartInfo>? = null
+)
+private data class MultiPartInfo(
+    val content: String? = null,
+    val contentDisposition: String? = null,
+    val contentLength: String? = null,
+    val contentType: String? = null
+)
+
+private fun processPart(part: String): MultiPartInfo {
+
+    val partLines = part.split("\r\n").let {
+        // part would look like "\r\nHeader-Name: Header-Value\r\n...\r\n\r\nbody\r\n"
+        // Here we're removing the first and last part as we know that are empty and
+        it.subList(1, it.size - 1)
+    }
+
+    val partLinesIterator = partLines.iterator()
+    val headers = mutableMapOf<String, String>()
+    for (headerLine in partLinesIterator) {
+        if (headerLine.isEmpty()) {
+            break
+        } else {
+            val (name, value) = headerLine.split(": ", limit = 2)
+            headers[name] = value
+        }
+    }
+
+    val content = partLinesIterator.asSequence().joinToString("\r\n")
+    return MultiPartInfo(
+        content = content,
+        contentDisposition = headers["Content-Disposition"],
+        contentType = headers["Content-Type"],
+        contentLength = headers["Content-Length"]
+    )
+}
+
+private val RecordedRequest.fileApiStats: FileApiStats
+    get() {
+        /**
+         * requestBody has to be compliant to the HTTP Specifications for MultiPart (RFC 2387)
+         *
+         * This means that the body will look like
+         *      --<BOUNDARY>\r\n
+         *      <Content1 Header Name>: <Content1 Header Value>\r\n
+         *      [<Content1 Header Name>: <Content1 Header Value>\r\n]
+         *      \r\n
+         *      <Content1>\r\n
+         *      --<BOUNDARY>\r\n
+         *      <Content# Header Name>: <Content# Header Value>\r\n
+         *      [<Content# Header Name>: <Content# Header Value>\r\n]
+         *      \r\n
+         *      <Content#>\r\n
+         *      --<BOUNDARY>--\r\n
+         *      \r\n
+         */
+        val requestBody = this.body.readUtf8()
+
+        val boundaryLine = requestBody.splitToSequence("\r\n").first()
+
+        val errors = mutableListOf<String>()
+        val bodyParts = mutableListOf<String>()
+
+        val response = FileApiStats()
+
+        if (!boundaryLine.startsWith("--")) {
+            errors.add("Boundary should start with '--'")
+        } else {
+            response.boundary = boundaryLine.substring(startIndex = 2)
+            bodyParts.addAll(requestBody.split(boundaryLine))
+            if (!bodyParts.last().startsWith("--")) {
+                errors.add("Start and End Boundaries are different")
+            } else if (bodyParts.last().substring(startIndex = 2) != "\r\n") {
+                errors.add("The last line of the body has to be empty")
+            }
+        }
```

Comment:
I'd remove the whole `errors` field at all and just call `fail()` here.
It's pointless to collect errors if you know that you want to let the test fail as soon as you have one.